### PR TITLE
[RFR] Fix memory adapter reset method

### PR DESCRIPTION
**Pull request by jpetitcolas**

### CODE REVIEW

File: src/adapters/memory.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default () => {
         const create = dateNow => {
             dictionary[id] = {
                 remaining: max,
-                reset: (dateNow + duration) / 1000,
+                reset: (dateNow + duration) / 1000 | 0,
```
Comment (jpetitcolas) [inline src/adapters/memory.js]: Remove `| 0`

File: src/adapters/memory.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ export default () => {
 
         const reset = (data, dateNow) => {
             data.remaining = max;
-            data.reset = data.reset * 1000 - dateNow;
+            data.reset = (dateNow + duration) / 1000 | 0;
```
Comment (jpetitcolas) [inline src/adapters/memory.js]: Issue was here. `reset` should be the next reset date. After resetting, `reset` becomes negative, causing some troubles.

File: src/adapters/memory.spec.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ describe('MemoryAdapter', () => {
         describe('when the duration is exceeded', function testDuration() {
             it('should reset', done => {
                 this.timeout(5000);
-                const adapter = adapterFactory()('foo', 2, 2000);
+                const adapter = adapterFactory()('foo', 2, 1000);
```
Comment (jpetitcolas) [inline src/adapters/memory.spec.js:75]: Reduced duration for faster tests.

**Comment by Kmaschta** [comment]
:+1:

**Pull request merged**